tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -4266,11 +4266,14 @@ def __init__(self, input_dataset, buffer_size, slack_period=None):
       buffer_size = -1  # This is the sentinel for auto-tuning.
     self._buffer_size = ops.convert_to_tensor(
         buffer_size, dtype=dtypes.int64, name="buffer_size")
-    variant_tensor = gen_dataset_ops.prefetch_dataset(
-        input_dataset._variant_tensor,  # pylint: disable=protected-access
-        buffer_size=self._buffer_size,
-        slack_period=slack_period,
-        **self._flat_structure)
+   
```

Comment:
Please move this device scope to `prefetching_ops.py`, `prefetch_to_device` implementation.

Reply:
Not sure it's doable actually. And i think it makes more sense the closer to the prefetching OP as possible.

For instance if someone is reimplementing (for any reason) `def prefetch_to_device(device, buffer_size=None):`, then at least like it is currently implemented we are still good